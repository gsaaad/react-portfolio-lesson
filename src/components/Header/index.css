.Navbar {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  /* font-size: 18px; */
  color: red;
}
ul {
  display: flex;
  flex-direction: row;
  list-style: none;
}
