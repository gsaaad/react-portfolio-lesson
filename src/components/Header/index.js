import React, { useEffect } from "react";
import "./index.css";
function Header(props) {
  const { category = [], currentCategory, setCurrentCategory } = props;

  useEffect(() => {
    document.title = currentCategory.name;
  }, [currentCategory]);
  return (
    <div>
      <header>
        <nav className="Navbar">
          <h1>GS</h1>
          <ul>
            {category.map((eachCategory) => (
              <li
                onClick={() => {
                  setCurrentCategory(eachCategory);
                }}
              >
                {eachCategory.name}
              </li>
            ))}

            {/* <li>
              <a href="/">About</a>
            </li>
            <li>
              <a href="https://www.google.com/">Portfolio</a>
            </li>
            <li>
              <a href="https://www.google.com/">Contact</a>
            </li>
            <li>
              <a href="https://www.google.com/">Resume</a>
            </li> */}
          </ul>
        </nav>
      </header>
    </div>
  );
}

export default Header;
