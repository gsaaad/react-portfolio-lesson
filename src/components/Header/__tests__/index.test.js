import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "..";

// go through each test and after, clean up and dispose of anything we aren't using
afterEach(cleanup);

describe("Header component", () => {
  // render Header test

  // first test
  it("renders", () => {
    render(<Header />);
  });
});

describe("Matches snapshot DOM node structure", () => {
  // use asFragment to return snapshot of DOM

  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
