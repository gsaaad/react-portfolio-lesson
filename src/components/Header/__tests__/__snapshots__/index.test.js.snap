// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div>
    <header>
      <nav
        class="Navbar"
      >
        <h1>
          GS
        </h1>
        <ul>
          <li>
            <a
              href="https://www.google.com/"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="https://www.google.com/"
            >
              Portfolio
            </a>
          </li>
          <li>
            <a
              href="https://www.google.com/"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="https://www.google.com/"
            >
              Resume
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</DocumentFragment>
`;
